pyro/poutine/escape_poutine.py

```diff
@@ -32,6 +31,27 @@ def _pyro_sample(self, msg):
         """
         if self.escape_fn(msg):
             msg["done"] = True
-            raise NonlocalExit(msg)
-        else:
-            return super(EscapePoutine, self)._pyro_sample(msg)
+            msg["stop"] = True
+
+            def cont(m):
+                # for enumeration, we need a value at the site to get the shape
```

Comment:
Why not just call `.shape()`?

Reply:
I wasn't sure if that was correct, but if so I can change it